app/controllers/TransferAgreementController.scala

```diff
@@ -43,9 +38,22 @@ class TransferAgreementController @Inject()(val controllerComponents: SecurityCo
         .verifying("All records must be open", b => b)
     )(TransferAgreementData.apply)(TransferAgreementData.unapply)
   )
+  private val options: Seq[(String, String)] = Seq("Yes" -> "true", "No" -> "false")
+  implicit val language: Lang = langs.availables.head
+  private val addTransferAgreementClient = graphqlConfiguration.getClient[AddTransferAgreement.Data, AddTransferAgreement.Variables]()
+
+  def transferAgreement(consignmentId: UUID): Action[AnyContent] = secureAction.async { implicit request: Request[AnyContent] =>
+    val consignmentStatusService = new ConsignmentStatusService(graphqlConfiguration)
 
-  def transferAgreement(consignmentId: UUID): Action[AnyContent] = consignmentExists(consignmentId) { implicit request: Request[AnyContent] =>
-    Ok(views.html.transferAgreement(consignmentId, transferAgreementForm, options))
+    for (consignmentStatus <- consignmentStatusService.consignmentStatus(consignmentId, request.token.bearerAccessToken)) yield {
+      val transferAgreementStatus: String = consignmentStatus.flatMap(_.transferAgreement).getOrElse("")
+      if (transferAgreementStatus == "Completed") {
+        Ok(views.html.transferAgreementAlreadyCompleted(consignmentId, transferAgreementForm, options))
+          .withHeaders("Cache-Control" -> "no-store, must-revalidate")
+      } else {
+        Ok(views.html.transferAgreement(consignmentId, transferAgreementForm, options))
```

Comment:
This version also needs the `Cache-Control` header. Otherwise the user's browser will cache this version, and it won't even bother to request the updated version when the user goes back to the page.